bundles/persistence/org.openhab.persistence.dynamodb/java/org/openhab/persistence/dynamodb/internal/DynamoDBPersistenceService.java

```diff
@@ -353,17 +353,31 @@ public void store(Item item, String alias) {
      * @param filter
      * @return DynamoDBQueryExpression corresponding to the given FilterCriteria
      */
-    private DynamoDBQueryExpression<DynamoDBItem<?>> createQueryExpression(FilterCriteria filter) {
-        final DynamoDBStringItem itemHash = new DynamoDBStringItem(filter.getItemName(), null, null);
+    private DynamoDBQueryExpression<DynamoDBItem<?>> createQueryExpression(Class<? extends DynamoDBItem<?>> dtoClass,
+            FilterCriteria filter) {
+        DynamoDBItem<?> item = getDynamoDBHashKey(dtoClass, filter.getItemName());
         final DynamoDBQueryExpression<DynamoDBItem<?>> queryExpression = new DynamoDBQueryExpression<DynamoDBItem<?>>()
-                .withHashKeyValues(itemHash).withScanIndexForward(filter.getOrdering() == Ordering.ASCENDING)
+                .withHashKeyValues(item).withScanIndexForward(filter.getOrdering() == Ordering.ASCENDING)
```

Comment:
This was the only change required by the new SDK. 

Previously a `DynamoDBStringItem` was always as a placeholder to enter the item name to the `DynamoDBQueryExpression`. New SDK uses the class in introspection and fails in the integration tests with number items (`DynamoDBBigDecimalItem`).

Now we construct new DTO class using the empty constructor, and set the name. This happens in `getDynamoDBHashKey`